### ISSUE TITLE
Change the exception type when parsing bad data

### ISSUE THREAD

**Issue by jleclanche** [python-pillow/Pillow#1643]
Right now PIL/Pillow recommends raising SyntaxError when dealing with bad data. Example in `PaletteFile`:

```python
            if len(s) > 100:
                raise SyntaxError("bad palette file")
```

**Comment by radarhere** [comment]
Hi. Thanks for contributing with this suggestion. My only input here is that there are many `SyntaxError` exceptions thrown throughout the image plugins, so if done properly, this is actually a bigger change than simply PaletteFile.

**Comment by jleclanche** [comment]
I agree, yes.

**Comment by meawoppl** [comment]
Each encoder should catch and homogenize potential errors thrown to this more canonical form.

**Comment by aclark4life** [comment]
@jleclanche You want to take a whack at this one?

**Comment by jleclanche** [comment]
@aclark4life yeah sure, but this is quite the backwards incompatible change. fine with that?

**Comment by wiredfool** [comment]
If only we had looked at a list of things to do for a breaking change release

**Comment by Cibai000** [comment]
release

**Comment by radarhere** [comment]
#7425 would also like a custom exception instead of an `OSError` raised in WebPImagePlugin.
